core/base-service/base-json.js

```diff
@@ -1,13 +1,42 @@
+/**
+ * @module
+ */
+
 'use strict'
 
 const BaseService = require('./base')
 const { parseJson } = require('./json')
 
+/**
+ * Services which query a JSON endpoint should extend BaseJsonService
+ *
+ * @abstract
+ */
 class BaseJsonService extends BaseService {
+  /**
+   * Parse data from JSON endpoint
+   *
+   * @param {string} buffer JSON repsonse from upstream API
+   * @returns {object} Parsed response
+   */
   _parseJson(buffer) {
     return parseJson(buffer)
   }
 
+  /**
+   * Request data from an upstream API serving JSON,
+   * parse it and validate against a schema
+   *
+   * @param {object} attrs Refer to individual attrs
+   * @param {Joi} attrs.schema Joi schema to validate the response against
+   * @param {string} attrs.url URL to request
+   * @param {object} [attrs.options={}] Options to pass to request. See
+   *    [documentation](https://github.com/request/request#requestoptions-callback)
+   * @param {object} [attrs.errorMessages={}] Key-value map of status codes
+   *    and custom error messages e.g: `{ 404: 'package not found' }`
```

Comment:
Since BaseService injects some defaults it would be useful to mention them. (This isn't blocking, just a suggested improvement.)